packages/slate/src/models/point.js

```diff
@@ -120,6 +120,79 @@ class Point extends Record(DEFAULTS) {
     return !this.isSet
   }
 
+  /**
+   * Check whether the point is after a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAfterRange(range) {
+    if (this.isUnset) return false
+    const is =
+      (this.key === range.end.key && this.offset > range.end.offset) ||
+      PathUtils.compare(this.path, range.end.path) === 1
+    return is
+  }
+
+  /**
+   * Check whether the point is at the end of a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAtEndOfRange(range) {
+    if (this.isUnset) return false
+    const is = this.key === range.end.key && this.offset === range.end.offset
+    return is
+  }
+
+  /**
+   * Check whether the point is at the start of a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isAtStartOfRange(range) {
+    if (this.isUnset) return false
+    const is =
+      this.key === range.start.key && this.offset === range.start.offset
+    return is
+  }
+
+  /**
+   * Check whether the point is before a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isBeforeRange(range) {
+    if (this.isUnset) return false
+    const is =
+      (this.key === range.start.key && this.offset < range.start.offset) ||
+      PathUtils.compare(this.path, range.start.path) === -1
+    return is
+  }
+
+  /**
+   * Check whether the point is inside a `range`.
+   *
+   * @return {Boolean}
+   */
+
+  isInRange(range) {
+    if (this.isUnset) return false
+
+    const afterOrAtStart =
```

Comment:
Could you refactor these to use the other methods defined in this pull request to make it a bit easier to follow and reduce code?

Reply:
Of course, I wasn't sure if I could use the other methods because it would lead to potentially more calls to `compare` but it would be easier to understand and maintain. I'll fix it.